ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,488 @@
+
+    return _traced_method
+
+
+def _inject_tracing_async_actor_method(method: Callable[..., Any]) -> Any:
+    async def _traced_async_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+
+        if not tracer or (_dd_trace_ctx is None and tracer.current_span() is None):
+            return await method(self, *args, **kwargs)
+
+        with _trace_actor_method(self, method, _dd_trace_ctx, *args, **kwargs):
+            return await method(self, *args, **kwargs)
+
+    return _traced_async_method
+
+
+def inject_tracing_into_actor_class(wrapped, instance, args, kwargs):
+    if not tracer:
+        return wrapped(*args, **kwargs)
+
+    cls = wrapped(*args, **kwargs)
+    module_name = str(cls.__module__)
+    class_name = str(cls.__name__)
+
+    # Skip tracing for certain ray modules
+    if module_name.startswith("ray.dag") or module_name.startswith("ray.experimental"):
+        return cls
+
+    # Determine if the class is a JobSupervisor
+    is_job_supervisor = f"{module_name}.{class_name}" == "ray.dashboard.modules.job.job_supervisor.JobSupervisor"
+    # We do not want to instrument ping and polling to remove noise
+    methods_to_ignore = {"ping", "_polling"} if is_job_supervisor else set()
+
+    methods = inspect.getmembers(cls, is_function_or_method)
+    for name, method in methods:
+        if name in methods_to_ignore:
+            continue
+
+        if (
+            is_static_method(cls, name)
+            or is_class_method(method)
+            or inspect.isgeneratorfunction(method)
+            or inspect.isasyncgenfunction(method)
+            or name == "__del__"
+        ):
+            continue
+
+        method.__signature__ = _inject_dd_trace_ctx_kwarg(method)
+
+        # Special handling for the run method in JobSupervisor
+        if is_job_supervisor and name == "run":
+            wrapped_method = wraps(method)(_job_supervisor_run_wrapper(method))
+        elif is_job_supervisor and name == "_exec_entrypoint":
+            wrapped_method = wraps(method)(_exec_entrypoint_wrapper(method))
+        else:
+            if inspect.iscoroutinefunction(method):
+                wrapped_method = wraps(method)(_inject_tracing_async_actor_method(method))
+            else:
+                wrapped_method = wraps(method)(_inject_tracing_actor_method(method))
+
+        setattr(cls, name, wrapped_method)
+    return cls
+
+
+async def traced_end_job(wrapped, instance, args, kwargs):
+    if not tracer:
+        return await wrapped(*args, **kwargs)
+
+    result = await wrapped(*args, **kwargs)
+
+    job_id = args[0]
+    job_info = await instance._job_info_client.get_info(job_id)
+    stop_long_running_job(job_id, job_info)
+
+    return result
+
+
+def patch():
+    if getattr(ray, "_datadog_patch", False):
+        return
+
+    ray._datadog_patch = True
+
+    tracer._span_aggregator.user_processors.append(RayTraceProcessor())
```

Comment:
Using dd_processors makes more sense here. user_processors are reset when tracer.configure(...) is called by customers. Also I left a note above about how we can get rid of this processor altogether. Since we are providing a ray instrumentation we have full control over how ray spans are created, we can set these tags without a processor

```suggestion
    tracer._span_aggregator.dd_processors.append(RayTraceProcessor())
```